it/src/main/java/com/google/cloud/teleport/it/DataGenerator.java

```diff
@@ -0,0 +1,153 @@
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.auth.Credentials;
+import com.google.cloud.teleport.it.dataflow.DataflowClient;
+import com.google.cloud.teleport.it.dataflow.DataflowClient.JobInfo;
+import com.google.cloud.teleport.it.dataflow.DataflowClient.JobState;
+import com.google.cloud.teleport.it.dataflow.DataflowClient.LaunchConfig;
+import com.google.cloud.teleport.it.dataflow.DataflowOperator;
+import com.google.cloud.teleport.it.dataflow.DataflowOperator.Result;
+import com.google.cloud.teleport.it.dataflow.DefaultDataflowClient;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+/** Helper class for starting a Streaming Data generator dataflow template job. */
+public class DataGenerator {
+  private static final String SPEC_PATH =
+      "gs://dataflow-templates/latest/flex/Streaming_Data_Generator";
+  private static final String PROJECT = TestProperties.project();
+  private static final String REGION = TestProperties.region();
+  private static final Credentials CREDENTIALS = TestProperties.googleCredentials();
+  private final LaunchConfig dataGeneratorOptions;
+  private final DataflowClient dataflowClient;
+  private final DataflowOperator dataflowOperator;
+
+  private DataGenerator(Builder builder) {
+    dataflowClient = DefaultDataflowClient.builder().setCredentials(CREDENTIALS).build();
+    dataflowOperator = new DataflowOperator(dataflowClient);
+    this.dataGeneratorOptions =
+        LaunchConfig.builder(builder.getJobName())
+            .setSpecPath(SPEC_PATH)
+            .setParameters(builder.getParameters())
+            .build();
+  }
+
+  public static DataGenerator.Builder builder(String jobName) {
+    return new DataGenerator.Builder(jobName);
+  }
+
+  /**
+   * Executes the data generator using the config provided. If messageLimit is provided, we wait
+   * until the data generator finishes or we reach timeout. If a messageLimit is not provided we
+   * wait until timeout and cancel the data generator.
+   *
+   * <p>Note: This is a blocking call. For backlog tests, start the Data generator before executing
+   * the template under test. For testing a pipeline against live incoming data, execute the data
+   * generator after starting the pipeline under test.
+   *
+   * @param timeout time to wait before cancelling the data generator.
+   * @throws IOException if any errors are encountered.
+   */
+  public void execute(Duration timeout) throws IOException {
+    JobInfo dataGeneratorJobInfo =
+        dataflowClient.launchFlexTemplate(PROJECT, REGION, dataGeneratorOptions);
+    assertThat(dataGeneratorJobInfo.state()).isIn(JobState.ACTIVE_STATES);
+    DataflowOperator.Config config = createConfig(dataGeneratorJobInfo, timeout);
+    // check if the job will be BATCH or STREAMING
+    Result dataGeneratorResult;
+    if (dataGeneratorOptions.parameters().containsKey("messagesLimit")) {
```

Comment:
Repeated a few times, suggest extracting to a constant

Reply:
Done.